### PR TITLE
Bug fixes

### LINKED ISSUE #110: Sometimes the logs contain messages from other repos
see: https://s3.amazonaws.com/cla-bot/ColinEberhardt-78527442-a6ac-47ec-ac4b-e24b8fb86729

It contains log messages from two different repos:

```
2019-02-02T14:10:08.483Z INFO Checking CLAs for pull request https://api.github.com/repos/evolvedbinary/pebble-extension/pulls/39
2019-02-02T14:10:08.484Z INFO Bot installed as an integration, obtaining installation token
2019-02-02T14:10:08.519Z INFO API Request https://api.github.com/installations/426748/access_tokens
2019-02-02T14:10:08.633Z INFO Attempting to obtain organisation level .clabot file URL
2019-02-02T14:10:08.634Z INFO API Request https://api.github.com/repos/evolvedbinary/clabot-config/contents/.clabot
2019-02-02T14:10:08.698Z INFO Organisation configuration not found, resolving .clabot URL at project level
2019-02-02T14:10:08.698Z INFO API Request https://api.github.com/repos/evolvedbinary/pebble-extension/contents/.clabot
2019-02-02T14:10:08.778Z INFO Obtaining .clabot configuration file from https://raw.githubusercontent.com/evolvedbinary/pebble-extension/master/.clabot
2019-02-02T14:10:08.778Z INFO API Request https://raw.githubusercontent.com/evolvedbinary/pebble-extension/master/.clabot
2019-02-02T14:10:08.952Z INFO Obtaining the list of commits for the pull request
2019-02-02T14:10:08.952Z INFO API Request https://api.github.com/repos/evolvedbinary/pebble-extension/pulls/39/commits
2019-02-02T14:10:09.068Z INFO Total Commits: 1, checking CLA status for committers
2019-02-02T14:10:09.068Z INFO All contributors have a signed CLA, adding success status to the pull request and a label
2019-02-02T14:10:09.069Z INFO API Request https://api.github.com/repos/evolvedbinary/pebble-extension/issues/39/labels
2019-02-02T14:10:09.217Z INFO API Request https://api.github.com/repos/evolvedbinary/pebble-extension/issues/39/labels
2019-02-02T14:10:09.218Z INFO API Request https://api.github.com/repos/evolvedbinary/pebble-extension/statuses/1074c8402c52811f19e80fb23c25081199e3932b

2019-02-02T14:49:38.054Z INFO Checking CLAs for pull request https://api.github.com/repos/ColinEberhardt/clabot-prod-test/pulls/10
2019-02-02T14:49:38.054Z INFO Bot installed as an integration, obtaining installation token
2019-02-02T14:49:38.059Z INFO API Request https://api.github.com/installations/39490/access_tokens
2019-02-02T14:49:38.110Z INFO Attempting to obtain organisation level .clabot file URL
2019-02-02T14:49:38.110Z INFO API Request https://api.github.com/repos/ColinEberhardt/clabot-config/contents/.clabot
2019-02-02T14:49:38.172Z INFO Organisation configuration found!
2019-02-02T14:49:38.172Z INFO Obtaining .clabot configuration file from https://raw.githubusercontent.com/ColinEberhardt/clabot-config/master/.clabot
2019-02-02T14:49:38.172Z INFO API Request https://raw.githubusercontent.com/ColinEberhardt/clabot-config/master/.clabot
2019-02-02T14:49:38.676Z INFO Obtaining the list of commits for the pull request
2019-02-02T14:49:38.676Z INFO API Request https://api.github.com/repos/ColinEberhardt/clabot-prod-test/pulls/10/commits
2019-02-02T14:49:38.849Z INFO Total Commits: 1, checking CLA status for committers
2019-02-02T14:49:38.866Z INFO API Request https://gist.githubusercontent.com/ColinEberhardt/293439a97af26a64f8d588ca9e242fad/raw/eab0ce188744e6aa757eaa1b559e2561e1d12e1a/contributors?colineberhardt
2019-02-02T14:49:38.891Z INFO All contributors have a signed CLA, adding success status to the pull request and a label
2019-02-02T14:49:38.891Z INFO API Request https://api.github.com/repos/ColinEberhardt/clabot-prod-test/issues/10/labels
2019-02-02T14:49:39.021Z INFO API Request https://api.github.com/repos/ColinEberhardt/clabot-prod-test/issues/10/labels
2019-02-02T14:49:39.022Z INFO API Request https://api.github.com/repos/ColinEberhardt/clabot-prod-test/statuses/928c73218299bdb16d8df77a07852fd8943d5766
```

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Fixes #110 #116

### CODE REVIEW

**Comment by maoo** [comment]
LGTM!

File: src/index.js [hunk 1]
```diff
@@ -135,6 +135,25 @@ exports.handler = constructHandler(async webhook => {
     addRecheckComment
   } = applyToken(token);
 
+  logger.info("Obtaining the list of commits for the pull request");
```
Comment (ColinEberhardt) [inline src/index.js:138]: This code was moved from further up the file as it is needed to obtain the SHA for the most recent commit in order to report errors relating to malformed JSON via the status API

File: src/index.js [hunk 2]
```diff
@@ -156,31 +175,14 @@ exports.handler = constructHandler(async webhook => {
   const config = await getFile(orgConfig);
 
   if (!is.json(config)) {
+    logger.error("The .clabot file is not valid JSON");
+    await setStatus(webhook, headSha, "error", logFile);
```
Comment (ColinEberhardt) [inline src/index.js:179]: report malformed JSON via the status API (i.e. visible to end users of the bot)

File: src/logger.js [hunk 3]
```diff
@@ -31,6 +31,8 @@ const logger = {
     logMessage("ERROR", message, detail);
   },
   logFile(filename) {
+    loggedMessages.length = [];
```
Comment (ColinEberhardt) [inline src/logger.js:34]: Clear the (shared!) global state before the bot executes. This fixes the bug where log messages from separate invocations were both ending up in the same file.

I've just spotted that the JavaScript code here is a bit mangled - it should be `arr.length = 0` or `arr = []` - I seem to have merged both approaches. However, JavaScript helpfully type converts `[]` to `0` so this actually works!

I'll fix this next time I have some more features / bugs

File: src/logger.js [hunk 4]
```diff
@@ -52,7 +54,7 @@ const logger = {
         .putObject({
           Body: detailedLoggedMessages.join("\r\n"),
           Bucket: process.env.LOGGING_BUCKET,
-          Key: `${id}-DEBUG`,
+          Key: `${logFile}-DEBUG`,
```
Comment (ColinEberhardt) [inline src/logger.js:57]: This fixes another bug, the 'detailed' log messages were incorrectly names

**Review by ColinEberhardt — COMMENTED** [review]
👍

**Pull request merged**